src/include/sof/audio/codec_adapter/codec/generic.h

```diff
@@ -0,0 +1,32 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2020 Intel Corporation. All rights reserved.
+ *
+ *
+ * \file generic.h
+ * \brief Generic Codec API header file
+ * \author Marcin Rajwa <marcin.rajwa@linux.intel.com>
+ *
+ */
+
+#ifndef __SOF_AUDIO_CODEC_GENERIC__
+#define __SOF_AUDIO_CODEC_GENERIC__
+
+#include <sof/audio/component.h>
+
+/*****************************************************************************/
+/* Codec generic data types						     */
+/*****************************************************************************/
+enum ca_state {
+	PP_STATE_DISABLED = 0,
```

Comment:
Can we have doxygen comments for all the new API structs and enums. This code will be used by lots of external folks so it need to be documented.